compose.py

```diff
@@ -0,0 +1,297 @@
+'''.format(backend, backend)
+        else:
+            log("Cannot create container from unsupported backend: " + backend)
+
+    with open(os.path.join(ddir, dockerfile_name), "a") as dfile:
+        dfile.write(df)
+
+
+def add_requested_repoagents(ddir, dockerfile_name, repoagents):
+    df = "#  Copying over repoagents \n"
+    for ra in repoagents:
+        if (ra in EXAMPLE_REPOAGENTS):
+            df += '''COPY --chown=1000:1000 --from=full /opt/tritonserver/repoagents/{} /opt/tritonserver/repoagents/{}    
+'''.format(ra, ra)
+        else:
+            log("Cannot create container from unsupported repoagent: " + ra)
+
+    with open(os.path.join(ddir, dockerfile_name), "a") as dfile:
+        dfile.write(df)
+
+
+
+# Install dependencies and run entrypoint script
+def end_gpu_dockerfile(ddir, dockerfile_name):
+    # TODO: remove libnvidia-ml-dev after 21.06 is launched
+    df = '''
+# Ensure apt-get won't prompt for selecting options
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+            libnvidia-ml-dev \
+            libre2-dev \
+            libb64-dev \
+            pciutils && \
+    rm -rf /var/lib/apt/lists/*
+'''
+
+    # Add dependencies needed for python backend
+    if 'python' in FLAGS.backend:
+        df += '''
+# python3, python3-pip and some pip installs required for the python backend
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+         python3 \
+         python3-pip && \
+    pip3 install --upgrade pip && \
+    pip3 install --upgrade wheel setuptools && \
+    pip3 install --upgrade grpcio-tools grpcio-channelz numpy && \
+    rm -rf /var/lib/apt/lists/*
+'''
+    df += '''
+# Add entrypoint script
+WORKDIR /opt/tritonserver
+COPY --chown=1000:1000 --from=full /opt/tritonserver/nvidia_entrypoint.sh .
+ENTRYPOINT ["/opt/tritonserver/nvidia_entrypoint.sh"]
+'''
+    with open(os.path.join(ddir, dockerfile_name), "a") as dfile:
+        dfile.write(df)
+
+
+### Create container with docker build
+def build_docker_container(ddir, dockerfile_name, container_name):
+    p = subprocess.Popen(['docker', 'build', '-t', container_name, '-f', \
+        os.path.join(ddir, dockerfile_name), '.'])
+    p.wait()
+    fail_if(p.returncode != 0, 'docker build {} failed'.format(container_name))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    group_qv = parser.add_mutually_exclusive_group()
+    group_qv.add_argument('-q',
+                          '--quiet',
+                          action="store_true",
+                          required=False,
+                          help='Disable console output.')
+    group_qv.add_argument('-v',
+                          '--verbose',
+                          action="store_true",
+                          required=False,
+                          help='Enable verbose output.')
+    parser.add_argument('--build-name',
```

Comment:
There was a previous comment on this... Use --output-name and describe as "Name for the generated Docker image."